File: Python/contains-dupe-II.py
def containsNearbyDuplicate(nums, k):
    flag = False
    for i in range(len(nums)):
        for j in range(i+1,len(nums)):
            if nums[i] == nums[j] and abs(i-j) <= k:
                flag = True                    

    return flag
 
containsNearbyDuplicate([1,2,3,1,2,3],2)

File: C++/contains-dupe-II.cpp
#include <bits/stdc++.h> 
class Solution {
public:
    bool containsNearbyDuplicate(std::vector<int>& nums, int k) {
        std::map<int, int> hash_map;

        if (nums.size() == 1) return false;

        for (size_t i = 0; i < nums.size(); ++i)
        {
            if (hash_map.count(nums[i]) > 0 && std::abs(int(i - hash_map[nums[i]])) <= k) return true;

            hash_map[nums[i]] = i;
        }
        return false;
    }
};